plugin/evm/epochs_manager_test.go

```diff
@@ -0,0 +1,75 @@
+package evm
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testEpochs struct {
+	DetailsFunc func(epoch uint64) (EpochDetails, error)
+}
+
+func (t *testEpochs) Details(epoch uint64) (EpochDetails, error) {
+	return t.DetailsFunc(epoch)
+}
+
+func TestByTimestamp(t *testing.T) {
+	callsCount := 0
+	testEpochsMock := &testEpochs{
+		DetailsFunc: func(epoch uint64) (EpochDetails, error) {
+			callsCount++
+			epochDetails1 := EpochDetails{
+				StartTime: big.NewInt(time.Date(2022, time.January, 1, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+				EndTime:   big.NewInt(time.Date(2022, time.January, 2, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+			}
+			epochDetails2 := EpochDetails{
+				StartTime: big.NewInt(time.Date(2022, time.January, 2, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+				EndTime:   big.NewInt(time.Date(2022, time.January, 3, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+			}
+			epochDetails3 := EpochDetails{
+				StartTime: big.NewInt(time.Date(2022, time.January, 3, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+				EndTime:   big.NewInt(time.Date(2022, time.January, 4, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+			}
+			epochDetails4 := EpochDetails{
+				StartTime: big.NewInt(time.Date(2022, time.January, 4, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+				EndTime:   big.NewInt(time.Date(2022, time.December, 4, 0, 0, 0, 0, time.UTC).Unix()).Uint64(),
+			}
```

Comment:
These should not be defined in this func but instead be at the top of the testing function. This way the body of the function can focus on the actual logic.